pytensor/link/pytorch/dispatch/scalar.py

```diff
@@ -0,0 +1,241 @@
+def pytorch_funcify_ScalarOp(op, node, **kwargs):
+    """Return pytorch function that implements the same computation as the Scalar Op.
+
+    This dispatch is expected to return a pytorch function that works on Array inputs as Elemwise does,
+    even though it's dispatched on the Scalar Op.
+    """
+
+    nfunc_spec = getattr(op, "nfunc_spec", None)
+    if nfunc_spec is None:
+        raise NotImplementedError(f"Dispatch not implemented for Scalar Op {op}")
+
+    func_name = nfunc_spec[0]
+
+    pytorch_func = getattr(torch, func_name)
+
+    if len(node.inputs) > op.nfunc_spec[1]:
+        # Some Scalar Ops accept multiple number of inputs, behaving as a variadic function,
+        # even though the base Op from `func_name` is specified as a binary Op.
+        # This happens with `Add`, which can work as a `Sum` for multiple scalars.
+        pytorch_variadic_func = getattr(torch, op.nfunc_variadic, None)
+        if not pytorch_variadic_func:
+            raise NotImplementedError(
+                f"Dispatch not implemented for Scalar Op {op} with {len(node.inputs)} inputs"
+            )
+
+        def pytorch_func(*args):
+            return pytorch_variadic_func(
+                torch.stack(torch.broadcast_tensors(*args), axis=0), axis=0
+            )
+
+    return pytorch_func
+
+
+@pytorch_funcify.register(Cast)
+def pytorch_funcify_Cast(op, **kwargs):
+    def cast(x):
+        return torch.tensor(x).astype(op.o_type.dtype)
+
+    return cast
+
+
+@pytorch_funcify.register(Identity)
+def pytorch_funcify_Identity(op, **kwargs):
+    def identity(x):
+        return x
+
+    return identity
+
+
+@pytorch_funcify.register(Clip)
+def pytorch_funcify_Clip(op, **kwargs):
+    """Register the translation for the `Clip` `Op`."""
+
+    def clip(x, min, max):
+        return torch.clip(x, min, max)
+
+    return clip
+
+
+# @pytorch_funcify.register(Composite)
+# def pytorch_funcify_Composite(op, node, vectorize=True, **kwargs):
+#     jax_impl = pytorch_funcify(op.fgraph)
+
+#     if len(node.outputs) == 1:
+
+#         def composite(*args):
+#             return jax_impl(*args)[0]
+
+#     else:
+
+#         def composite(*args):
+#             return jax_impl(*args)
+
+#     return jnp.vectorize(composite)
+
+
+@pytorch_funcify.register(Second)
+def pytorch_funcify_Second(op, **kwargs):
+    def second(x, y):
+        _, y = torch.broadcast_tensors(x, y)
+        return y
+
+    return second
+
+
+@pytorch_funcify.register(GammaIncInv)
+def pytorch_funcify_GammaIncInv(op, **kwargs):
+    gammaincinv = try_import_tfp_jax_op(op, jax_op_name="igammainv")
+
+    return gammaincinv
+
+
+@pytorch_funcify.register(GammaIncCInv)
+def pytorch_funcify_GammaIncCInv(op, **kwargs):
+    gammainccinv = try_import_tfp_jax_op(op, jax_op_name="igammacinv")
+
+    return gammainccinv
+
+
+@pytorch_funcify.register(Erf)
+def pytorch_funcify_Erf(op, node, **kwargs):
+    def erf(x):
+        return torch.special.erf(x)
```

Comment:
is torch.special a thing? If so we could try to extend the default scalar op dispatch to handle these cases

Reply:
It is a module based on [scipy.speacial](https://docs.scipy.org/doc/scipy/reference/special.html)